src/wallet/wallet_tx_builder.h

```diff
@@ -331,20 +338,43 @@ class WalletTxBuilder {
      */
     CAmount DefaultDustThreshold() const;
 
+    ChangeAddress
+    GetChangeAddress(
+            CWallet& wallet,
+            const ZTXOSelector& selector,
+            SpendableInputs& spendable,
+            const Payments& resolvedPayments,
+            const TransactionStrategy& strategy,
+            bool afterNU5) const;
```

Comment:
I think we may be incurring unnecessary complexity by supporting tx creation in the before-NU5 case. I know it's used in tests, but it should never be used in production, because you have to be synced to get a reasonable Sapling or Orchard anchor.

Reply:
I had asked about this before, and I don’t really recall the reason for preserving this. I think maybe something related to forks that don’t support NU5 or something? Like, separation between the chain and the implementation.

Reply:
Ehh. I think you're right about the original motivation, but it depends how much complexity we're incurring. It is reasonable to spend a little complexity budget on chains that don't follow an upgrade, but IMHO, not very much and/or not for too long after our mainnet activation.

Reply:
On reflection the added complexity is not really that great, since we need to account for the fact that we can't use Sprout and Orchard in the same transaction, anyway.